client/src/components/ShowAllProjects.jsx

```diff
@@ -1,14 +1,25 @@
 import styled from "styled-components";
+import { useState } from "react";
+import MyButton from "./MyButton";
 
 
 
-function ShowAllProjects({allProjects}){
-    
+export default function ShowAllProjects({allProjects}){
+    const [details, setDetails] = useState([])
+
+    function Toggle(projectID){
+        const newDetails = [...allProjects]
+        newDetails[projectID].in_detail = newDetails[projectID].in_detail ? false : true
```

Comment:
@doemser just mentioned that you could also write `newDetails[projectID].in_detail = !newDetails[projectID].in_detail` instead of `newDetails[projectID].in_detail = newDetails[projectID].in_detail ? false : true`